### ISSUE TITLE
rediscluster.lua:69: module 'redis' not found:

### ISSUE THREAD

**Issue by vkozubal** [steve0511/resty-redis-cluster#2]
I have a failure in rediscluster.lua:69 
`local redis = require "redis"`
Where I am supposed to get that package ?

`2017/09/16 16:05:48 [error] 123#0: *9 lua entry thread aborted: runtime error: /app/vendor/resty-redis-cluster/lib/rediscluster.lua:69: module 'redis' not found:
	no field package.preload['redis']
	no file '/app/vendor/resty-redis-cluster/lib/redis.lua'
	no file '/usr/local/openresty/lualib/redis/lib/redis.lua'
	no file '/usr/local/openresty/lualib/redis.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/redis.lua'
	no file '/usr/local/openresty/site/lualib/redis.so'
	no file '/usr/local/openresty/lualib/redis.so'
	no file './redis.so'
	no file '/usr/local/lib/lua/5.1/redis.so'
	no file '/usr/local/openresty/luajit/lib/lua/5.1/redis.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'`

**Comment by steve0511** [comment]
Hi, have you generate redis_slot.so and configure in the lualib path?

**Comment by vkozubal** [comment]
No. That's the problem :) any docs ? 
I can't compile redis_slot.c due to missing header files. How can I get those?
I have configured `lua_package_path '/app/vendor/resty-redis-cluster/lib/?.lua;`

**Comment by steve0511** [comment]
sorry for there is a typo , gcc parser.c -fPIC -shared -o redis_slot.so -l /usr/local/include/luajit-2.1 

should be gcc redis_slot.c -fPIC -shared -o redis_slot.so -l /usr/local/include/luajit-2.1 

Please configure so file like: lua_package_cpath "/app/vendor/resty-redis-cluster/lib/?.so;";

**Comment by vkozubal** [comment]
now it says:
`2017/09/16 17:33:16 [error] 225#0: *16 lua entry thread aborted: runtime error: /app/vendor/resty-redis-cluster/lib/rediscluster.lua:4: module 'cjson' not found:
	no field package.preload['cjson']
	no file '/app/vendor/resty-redis-cluster/lib/cjson.lua'
	no file '/usr/local/openresty/lualib/resty/cjson.lua'
	no file '/usr/local/openresty/lualib/cjson/cjson.lua'
	no file '/usr/local/openresty/lualib/cjson.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/cjson.lua'
	no file '/app/vendor/resty-redis-cluster/lib/cjson.so'`

but on my file system I don't have cjson.lua file
`5da747f7be25:/# find / -name cjson*
/usr/local/openresty/lualib/cjson.so
/app/openresty-1.11.2.5/bundle/lua-cjson-2.1.0.5/lua/cjson
/app/openresty-1.11.2.5/build/lua-cjson-2.1.0.5/lua/cjson
/app/openresty-1.11.2.5/build/lua-cjson-2.1.0.5/cjson.so`

**Comment by vkozubal** [comment]
I this that was my bad. The error has changed since then. 
the latest error is
`2017/09/16 17:57:50 [error] 283#0: *25 [lua] rediscluster.lua:137: init_slots(): failed to create lock in initialization slot cache: dictionary not found, client: 127.0.0.1, server: localhost, request: "GET /redis HTTP/1.1", host: "localhost"
2017/09/16 17:57:50 [error] 283#0: *25 lua entry thread aborted: runtime error: /app/vendor/resty-redis-cluster/lib/rediscluster.lua:271: attempt to index local 'slots' (a nil value)
stack traceback:
coroutine 0:
	/app/vendor/resty-redis-cluster/lib/rediscluster.lua: in function 'handleCommandWithRetry'
	/app/vendor/resty-redis-cluster/lib/rediscluster.lua:383: in function 'get'
	content_by_lua(nginx.conf:82):18: in function <content_by_lua(nginx.conf:82):1>, client: 127.0.0.1, server: localhost, request: "GET /redis HTTP/1.1", host: "localhost"`

**Comment by vkozubal** [comment]
I was able to fix all problems by removing the 
`rediscluster.lua#4 local cjson = require "cjson"`
and by adding in the nginx.conf 
`lua_shared_dict redis_cluster_slot_locks 100k;`

**Comment by i2dcarrasco** [comment]
Hello,

I don't know if I'm doing something wrong or the filename is not right, but the rediscluster.lua file calls to librestyredisslot.so and the compiled filename is redis_slot.so. Are both the same file?, because I was not able to make it work and I've just copied redis_slot.so to  librestyredisslot.so and now looks like is working (I've to test it a bit but today I've no time).

Best regards.

**Comment by steve0511** [comment]
yes they are same, the doc should be updated as per the latest version, thanks.

**Comment by wangwenlon** [comment]
> No. That's the problem :) any docs ? I can't compile redis_slot.c due to missing header files. How can I get those? I have configured `lua_package_path '/app/vendor/resty-redis-cluster/lib/?.lua;`

where to find redis

> yes they are same, the doc should be updated as per the latest version, thanks.

please update the doc, I do not success  by following readme
